Common/AnatupleProducer.py

```diff
@@ -0,0 +1,185 @@
+import Common.ReportTools as ReportTools
+
+def DefineAndAppend(df, varToDefine, varToCall, colToSave):
+    df = df.Define(f"{varToDefine}", f"{varToCall}")
+    colToSave.append(varToDefine)
+    return df 
+
+def createAnatuple(inFile, outFile, period, sample, X_mass, snapshotOptions, isData=0):
+    Baseline.Initialize(True, True)
+    df = ROOT.RDataFrame("Events", inFile) 
+    df = df.Range(100)
+    df = df.Define("sample", f"static_cast<int>(SampleType::{sample})")
+    df = df.Define("period", f"static_cast<int>(Period::Run{period})") 
+    df = df.Define("X_mass", f"static_cast<int>({X_mass})")
+    
+    df = df.Define("GenPart_daughters", "GetDaughters(GenPart_genPartIdxMother)")
+    df = Baseline.RecoLeptonsSelection(df)
+    df = Baseline.RecoJetAcceptance(df)
+    df = Baseline.RecoHttCandidateSelection(df)
+    df = Baseline.RecoJetSelection(df) 
+    df = Baseline.DefineHbbCand(df)  
+    df = df.Define("channelId","static_cast<int>(httCand.channel())")
+    df = df.Define("is_data", f"{isData}") 
+    df = df.Define("lumi", "luminosityBlock")
+    df = df.Define("evt", "event")
+    deepTauScores= ["rawDeepTau2017v2p1VSe","rawDeepTau2017v2p1VSmu",\
+                "rawDeepTau2017v2p1VSjet", "rawDeepTau2018v2p5VSe", "rawDeepTau2018v2p5VSmu",\
+                "rawDeepTau2018v2p5VSjet", "decayMode"] 
+    JetObservables = ["hadronFlavour","partonFlavour", "particleNetAK4_B", "particleNetAK4_CvsB",\
+                    "particleNetAK4_CvsL","particleNetAK4_QvsG","particleNetAK4_puIdDisc"] 
+    colToSave = ["event","lumi","run","sample", "period", "X_mass","channelId", "is_data",\
+                "MET_pt", "MET_phi","PuppiMET_pt", "PuppiMET_phi",\
+                "DeepMETResolutionTune_pt", "DeepMETResolutionTune_phi","DeepMETResponseTune_pt", "DeepMETResponseTune_phi"] 
+ 
+    
+    for leg_idx in [0,1]:
+        df = DefineAndAppend(df, f"tau{leg_idx+1}_pt", f"httCand.leg_p4[{leg_idx}].Pt()", colToSave)
+        df = DefineAndAppend(df, f"tau{leg_idx+1}_eta", f"httCand.leg_p4[{leg_idx}].Eta()", colToSave)
+        df = DefineAndAppend(df,f"tau{leg_idx+1}_phi", f"httCand.leg_p4[{leg_idx}].Phi()", colToSave)
+        df = DefineAndAppend(df,f"tau{leg_idx+1}_mass", f"httCand.leg_p4[{leg_idx}].M()", colToSave)
+        df = DefineAndAppend(df,f"tau{leg_idx+1}_charge", f"httCand.leg_charge[{leg_idx}]", colToSave) 
+        df = df.Define(f"tau{leg_idx+1}_genMatch_class", f"GenRecoLepMatching(httCand.leg_p4[{leg_idx}],\
+                          GenPart_pdgId, GenPart_daughters,\
+                          GenPart_pt, GenPart_eta, GenPart_phi,\
+                          GenPart_mass, GenPart_statusFlags)")
+        df = DefineAndAppend(df,f"tau{leg_idx+1}_genMatch", f"static_cast<int>(tau{leg_idx+1}_genMatch_class);" , colToSave)
+        df = df.Define(f"tau{leg_idx+1}_recoJetMatchIdx", f"RecoTauMatching(httCand.leg_p4[{leg_idx}], Jet_p4)")
+        df = DefineAndAppend(df, f"tau{leg_idx+1}_iso", f"Float_t iso;\
```

Comment:
- use """ instead of having \ at the end of each line
- throw exception in the final `else`